### PR TITLE
Add support for Gentoo

### PR DESCRIPTION
**Pull request by devimc**

### CODE REVIEW

File: rootfs-builder/gentoo/rootfs_lib.sh [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+-pwquality -qrcode -repart -resolvconf sysv-utils -test -xkb -zstd"
+	local packages_optimizations="systemd multicall -ncurses -udev -debug \
+-X -selinux -doc -examples -nls -unicode -abi_x86_32 -abi_x86_x32"
+	local compiler_optimizations="-O3 -feliminate-unused-debug-types -pipe \
+-mtune=skylake -fexceptions -fstack-protector --param=ssp-buffer-size=32 \
+-Wl,--copy-dt-needed-entries -m64 -fasynchronous-unwind-tables \
+-Wp,-D_REENTRANT -ftree-loop-distribute-patterns -Wl,-z -Wl,now \
+-Wl,-z -Wl,relro -fno-semantic-interposition -ffat-lto-objects \
+-fno-signed-zeros -fno-trapping-math -fassociative-math -Wl,-sort-common \
+-Wl,--enable-new-dtags"
```
Comment (jodh-intel) [inline rootfs-builder/gentoo/rootfs_lib.sh:68]: Could you build this up line by line using an array or similar? That would make it much clearer, allow comments on any option and make diffs easier to read.

**Review by jodh-intel — COMMENTED** [review]
Thanks for raising @devimc.

**Comment by devimc** [comment]
@jodh-intel changes applied, thanks

/test

**Review by fidencio — APPROVED** [review]
lgtm, thanks @devimc.

**Comment by fidencio** [comment]
@devimc, labels are weird here. :-)
As we don't have this patch yet on 2.0, I've added the "forward-port-needed" tag.  I added "no-backport-needed", but we may want that on stable so, if that's the case, please, change that to "backport-needed".

**Comment by devimc** [comment]
@fidencio thanks

**Review by GabyCT — APPROVED** [review]

**Pull request merged**